packages/protocol/contracts/L1/TaikoData.sol

```diff
@@ -84,10 +82,17 @@ library TaikoData {
     }
 
     /// @dev Struct representing prover assignment
+    struct TierFee {
+        uint16 tier;
+        uint256 fee;
+    }
+
     struct ProverAssignment {
         address prover;
+        address feeToken;
+        TierFee[] tierFees;
         uint64 expiry;
-        bytes data;
+        bytes signature;
```

Comment:
With the rename of data -> signature it seems (based on the renaming only) it does not contain the 'tokenAmount' (for `IProver`) if custom ERC20. 
(This data tho still can be incorporate into the `signature` by `abi.encode(signature, tokenAmount)` so maybe best to use `data` otherwise it can be confusing)